Partitioner.hpp

```diff
@@ -163,8 +168,43 @@ class LIB_EXPORT Partitioner {
     // Vector of maps of neighbours to their halo sizes after partitioning
     std::vector<std::map<int, int>> _neighbours = std::vector<std::map<int, int>>(NNBRS);
 
+    // Vector of maps of neighbours to their halo start indices after partitioning
+    std::vector<std::map<int, int>> _halo_starts = std::vector<std::map<int, int>>(NNBRS);
+
     // Vector of maps of periodic neighbours to their halo sizes after partitioning
     std::vector<std::map<int, int>> _neighbours_p = std::vector<std::map<int, int>>(NNBRS);
+    //
+    // Vector of maps of periodic neighbours to their halo start indices after partitioning
+    std::vector<std::map<int, int>> _halo_starts_p = std::vector<std::map<int, int>>(NNBRS);
+
+private:
+    /*!
+     * @brief Check if 2 domains are neighbouring. If true, then domain 2 is
+     * the [edge] neighbour of domain 1, relative to domain 1.
+     * e.g., if domain 2 is to the right of domain 1, the following call will
+     * return true
+     * is_neighbour(d1, d2, RIGHT) == true
+     *
+     * @param d1 first domain
+     * @param d2 second domain
+     * @param edge LEFT, RIGHT, BOTTOM or TOP
+     * @param is_px are we looking for periodic neighbour in x-direction?
+     * @param is_py are we looking for periodic neighbour in y-direction?
+     * @return bool
+     */
+    bool is_neighbour(const Domain d1, const Domain d2, const Edge edge, const bool is_px = false,
+        const bool is_py = false);
+
+    /*!
+     * @brief Compute the start location of the halo for a given pair of domains. This should only
+     * be called if the two domains have already been confirmed as neighbours using is_neighbour().
```

Comment:
unfortunately it won't work as is. I wrote `halo_start` to be completely general (periodic/non-periodic) but `is_neighbour` can be run with or without periodic boundary cond's.

It is a nice idea, but I think it might be overkill for now.

I have modified the docs to make it clear the inputs are neighbouring domains.

```suggestion
     * @brief Compute the start location of the halo for a given pair of neighbouring domains.
```

Reply:
resolved in [f810ef9](https://github.com/nextsimhub/domain_decomp/pull/67/commits/f810ef9ce185763854a4b7684a42a34b4385439b)